reframe/frontend/cli.py

```diff
@@ -375,46 +375,27 @@ def main():
         # Locate and load checks
         try:
             checks_found = loader.load_all()
-        except OSError as e:
+        except OSError as oe:
             raise ReframeError from e
 
         # Filter checks by name
-        checks_matched = filter(
-            lambda c:
-            c if c.name not in options.exclude_names else None,
-            checks_found
-        )
+        checks_matched = filter(filters.have_not_name(options.exclude_names), checks_found)
 
         if options.names:
-            checks_matched = filter(
-                lambda c: c if c.name in options.names else None,
-                checks_matched
-            )
+            checks_matched = filter(filters.have_name(options.names), checks_matched)
 
         # Filter checks by tags
         user_tags = set(options.tags)
-        checks_matched = filter(
-            lambda c: c if user_tags.issubset(c.tags) else None,
-            checks_matched
-        )
+        checks_matched = filter(filters.have_tag(options.tags), checks_matched)
 
+        # TODO: should we move this in utils?
```

Comment:
Which one?

Reply:
Comment removed, it referred to the old filtering version